src/Validator/Constraints/MollieGatewayConfigValidator.php

```diff
@@ -26,54 +27,100 @@ final class MollieGatewayConfigValidator extends ConstraintValidator
 
     private const AMOUNT_LIMITS_FIELD = 'amountLimits';
 
-    private const FIELD_VALUE = 'value';
-
     public function validate(mixed $value, Constraint $constraint): void
     {
         Assert::isInstanceOf($constraint, MollieGatewayConfigValidatorType::class);
 
-        if ($value instanceof PersistentCollection) {
-            $this->validateAmounts($value, $constraint);
+        if (!$value instanceof PersistentCollection) {
+            return;
         }
-    }
 
-    private function validateAmounts(PersistentCollection $collection, MollieGatewayConfigValidatorType $constraint): void
-    {
-        $mollieGatewayConfigs = $collection->getSnapshot();
-        foreach ($mollieGatewayConfigs as $key => $mollieGatewayConfig) {
-            $amountLimits = $mollieGatewayConfig->getAmountLimits();
-            if (!$amountLimits) {
+        /** @var MollieGatewayConfigInterface[] $configs */
+        $configs = $value->getSnapshot();
+
+        foreach ($configs as $index => $config) {
+            $limits = $config->getAmountLimits();
+            if ($limits === null) {
```

Comment:
It could have stayed as it was
```suggestion
            $amountLimits = $config->getAmountLimits();
            if ($amountLimits === null) {
```